azext_iot/central/providers/device_template_provider.py

```diff
@@ -72,9 +73,25 @@ def list_device_templates(
             api_version=self._api_version,
         )
 
-        self._device_templates.update(
-            {template.id: template.raw_template for template in templates}
-        )
+        if compact:  # if asked for compact, just keep reduced info
+            for template in templates:
```

Comment:
This assumes that self._device_template won't have a key that templates does not have, correct? Would there be an edge case in which this assumption is invalid (through deletions and additions of templates)?

Reply:
Is the self._device_template used anywhere? I see that the code tries to keep it up to date but do not see any direct usage of it.

Reply:
the edge case is only possible if self._device_templates persists across commands. right now we removed all those cases and we just kept cache to be removed in future changes as you correctly pointed out.